modules/transformation.py

```diff
@@ -0,0 +1,609 @@
+        for entry in entries:
+            ret_request = {
+                "client": status_info["client"],
+                "date": status_info["date"],
+                "pageid": pageid,
+                "crawlid": status_info["crawlid"],
+                # we use this below for expAge calculation
+                "startedDateTime": utils.datetime_to_epoch(
+                    entry["startedDateTime"], status_info
+                ),
+                "time": entry["time"],
+                "_cdn_provider": entry.get("_cdn_provider"),
+                # amount response WOULD have been reduced if it had been gzipped
+                "_gzip_save": entry.get("_gzip_save"),
+            }
+
+            # REQUEST
+            try:
+                request = entry["request"]
+            except KeyError:
+                # TODO metric for failed parsing
+                logging.warning(
+                    f"Entry does not contain a request, status_info={status_info}, entry={entry}"
+                )
+                continue
+
+            url = request["url"]
+            (
+                request_headers,
+                request_other_headers,
+                request_cookie_size,
+            ) = utils.parse_header(
+                request["headers"], constants.ghReqHeaders, cookie_key="cookie"
+            )
+            ret_request.update(
+                {
+                    "method": request["method"],
+                    "httpVersion": request["httpVersion"],
+                    "url": url,
+                    "urlShort": url[:255],
+                    "reqHeadersSize": request.get("headersSize")
+                    if int(request.get("headersSize", 0)) > 0
+                    else None,
+                    "reqBodySize": request.get("bodySize")
+                    if int(request.get("bodySize", 0)) > 0
+                    else None,
+                    "reqOtherHeaders": request_other_headers,
+                    "reqCookieLen": request_cookie_size,
+                }
+            )
+
+            # RESPONSE
+            response = entry["response"]
+            status = response["status"]
+            ret_request.update(
+                {
+                    "status": status,
+                    "respHttpVersion": response["httpVersion"],
+                    "redirectUrl": response.get("url"),
+                    "respHeadersSize": response.get("headersSize")
+                    if int(response.get("headersSize", 0)) > 0
+                    else None,
+                    "respBodySize": response.get("bodySize")
+                    if int(response.get("bodySize", 0)) > 0
+                    else None,
+                    "respSize": response["content"]["size"],
+                }
+            )
+
+            # TODO revisit this logic - is this the right way to get extention, type, format from mimetype?
+            #  consider using mimetypes library instead https://docs.python.org/3/library/mimetypes.html
```

Comment:
I still like this approach and would be really interested to append some new columns to the tables with the standardized results for comparison.

Reply:
Agreed. When we last chatted, you mentioned adding additional fields to the schema so we could compare current and new outputs. Should we do that in this PR, or a separate one?

Reply:
Actually we agree to do this in a follow up PR